pmlc/dialect/pxa/transforms/tile.cc

```diff
@@ -36,6 +37,12 @@ void performTiling(AffineParallelOp op, llvm::ArrayRef<int64_t> tileSizes) {
   auto &outerLoopOps = outerBody->getOperations();
   innerLoopOps.splice(std::prev(innerLoopOps.end()), outerLoopOps,
                       std::next(outerLoopOps.begin(), 1), outerLoopOps.end());
+  auto innerIdxs = inner.getBody()->getArguments();
+  for (auto& innerOp : innerLoopOps) {
+    for (unsigned i = 0; i < outerIdxs.size(); ++i) {
+      innerOp.replaceUsesOfWith(outerIdxs[i], innerIdxs[i]);
```

Comment:
Hmm, I think this does not do the correct thing for any sort of deep loop structure (i.e. tiling the outer loops of an already 2 level affine.parallel).  I checked the implementation of op.replaceUsesWith and it doesn't walk into the regions inside of an Op.  I think you need to walk all the uses of the outerIdx and replace all of them with innerIdx, except the uses in the inner affine.for's definition.

Reply:
Let's add a deeply nested loop test case to check for this case.